From d9a75412c3e843890588a1e794f3014ee65e5619 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Sat, 14 Mar 2020 14:42:43 -0700
Subject: [PATCH] modemManager: Look for property on correct object

The SignalQuality property is defined on the GDBusProxy, not the modem
JS object.
Fix this runtime warning:

JS WARNING: [resource:///org/gnome/shell/misc/modemManager.js 252]: reference to undefined property "SignalQuality"
JS ERROR: TypeError: this.SignalQuality is undefined
_reloadSignalQuality@resource:///org/gnome/shell/misc/modemManager.js:252:34
_init@resource:///org/gnome/shell/misc/modemManager.js:234:14
NMDeviceModem@resource:///org/gnome/shell/ui/status/network.js:517:34
_deviceAdded@resource:///org/gnome/shell/ui/status/network.js:1755:27

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1097
---
 js/misc/modemManager.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/misc/modemManager.js b/js/misc/modemManager.js
index 21e49e81c5..4cc810bbb6 100644
--- a/js/misc/modemManager.js
+++ b/js/misc/modemManager.js
@@ -249,7 +249,7 @@ var BroadbandModem = GObject.registerClass({
     }
 
     _reloadSignalQuality() {
-        let [quality, recent_] = this.SignalQuality;
+        let [quality, recent_] = this._proxy.SignalQuality;
         this._setSignalQuality(quality);
     }
 
-- 
2.24.1

